PWGCF/Flow/Tasks/flowPbpbPikp.cxx

```diff
@@ -302,39 +352,78 @@ struct FlowPbpbPikp {
     histos.fill(HIST("hCent"), collision.centFT0C());
     fGFW->Clear();
     const auto cent = collision.centFT0C();
+
+    auto posSlicedTracks = posTracks->sliceByCached(aod::track::collisionId, collision.globalIndex(), cache);
+    auto negSlicedTracks = negTracks->sliceByCached(aod::track::collisionId, collision.globalIndex(), cache);
+
     float weff = 1, wacc = 1;
     int pidIndex;
-    for (auto const& track : tracks) {
-      double pt = track.pt();
-      histos.fill(HIST("hPhi"), track.phi());
-      histos.fill(HIST("hEta"), track.eta());
+
+    // resurrectParticle(posSlicedTracks, negSlicedTracks, kplusdaug, kminusdaug, Phimom, massKplus, massKminus, HIST("hPhiMass"));
+
+    for (auto const& trackA : posSlicedTracks) {
+      if (getNsigmaPID(trackA) != 2)
+        continue;
+      if (isFakeKaon(trackA))
+        continue;
+      auto trackAID = trackA.globalIndex();
+
+      for (auto const& trackB : negSlicedTracks) {
+        auto trackBID = trackB.globalIndex();
+        if (getNsigmaPID(trackB) != 2)
+          continue;
+        if (isFakeKaon(trackB))
+          continue;
+        if (trackAID == trackBID)
+          continue;
+
+        histos.fill(HIST("KplusTPC"), trackA.pt(), trackA.tpcSignal());
+        histos.fill(HIST("KminusTPC"), trackB.pt(), trackB.tpcSignal());
+
+        kplusdaug = ROOT::Math::PxPyPzMVector(trackA.px(), trackA.py(), trackA.pz(), massKplus);
+        kminusdaug = ROOT::Math::PxPyPzMVector(trackB.px(), trackB.py(), trackB.pz(), massKminus);
+        Phimom = kplusdaug + kminusdaug;
```

Comment:
Consider using the functionality in `Common/Core/RecoDecay.h`. It is by far more efficient specially if within a double inner loop as is the case